pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -136,11 +137,22 @@ func (r *ReconcileComplianceRemediation) Reconcile(request reconcile.Request) (r
 	}
 
 	var reconcileErr error
-	if remediationInstance.HasUnmetDependencies() {
+
+	if remediationInstance.HasUnmetDependencies() || remediationInstance.HasUnsetValues() || remediationInstance.HasValueRequired() {
```

Comment:
moving these to individual `if` branches would help with readability. e.g.

```
if remediationInstance.HasUnmetDependencies() {
...
} else if remediationInstance.HasUnsetValues() {
...
} else if remediationInstance.HasValueRequired() {
...
} else {
...
```

This will also help ensure that we don't depend on things set on `remediationInstance` which we shouldn't be modifying directly.

Reply:
This will skip the `if remediationInstance.HasValueRequired()` when there is unset-values

Reply:
I mean it will skip the next block if the previous ones are true, so the other blocks doesn't get handled

Reply:
I think I found a way around